### PR TITLE
Update e2e test as IMDSv1 is disabled by default

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?
This PR fixes E2E tests that rely on the hostname. IMDSv1 was disabled on the latest Ubuntu AMI.

### Describe how to test/QA your changes
Ran the failing e2e test.

### CODE REVIEW

**Comment by ogaca-dd** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

File: test/new-e2e/tests/agent-subcommands/hostname/hostname_ec2_nix_test.go [hunk 1]
```diff
@@ -47,13 +46,13 @@ func (v *linuxHostnameSuite) TestAgentConfigPreferImdsv2() {
...
 
 	metadata := client.NewEC2Metadata(v.T(), v.Env().RemoteHost.Host, v.Env().RemoteHost.OSFamily)
 	hostname := v.Env().Agent.Client.Hostname()
 
-	// Default configuration of hostname for EC2 instances is the resource-id
-	resourceID := metadata.Get("instance-id")
-	assert.Equal(v.T(), resourceID, hostname)
+	instanceID := metadata.Get("instance-id")
+	// As IMDSv1 is disabled by default, the hostname should not be the instance ID
+	assert.NotEqual(v.T(), instanceID, hostname)
```
Comment (pgimalac) [inline test/new-e2e/tests/agent-subcommands/hostname/hostname_ec2_nix_test.go:57]: I don't think we want to test that it doesn't default to something, that doesn't seem very useful 😄
Would you be able to update the test to make sure it defaults to whatever we expect it to ?

**Review by hush-hush — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]
Thanks !

**Pull request merged**